File: engine/scripts/lua/tests/test_map_metadata.lua
local assert = require("assert")

function test_map_metadata()
	add_cell(1, 2, 0)
	set_cell_metadata({ x = 1, y = 2, z = 0 }, { biome = "Forest", terrain = "Grass" })
	local meta = get_cell_metadata({ x = 1, y = 2, z = 0 })
	assert.equals(meta.biome, "Forest")
	assert.equals(meta.terrain, "Grass")
end

return { test_map_metadata = test_map_metadata }

File: engine_py/tests/test_map_metadata.py
def test_map_metadata(make_world):
    world = make_world()
    world.add_cell(1, 2, 0)
    world.set_cell_metadata(
        {"Square": {"x": 1, "y": 2, "z": 0}},
        {"biome": "Forest", "terrain": "Grass"},
    )
    meta = world.get_cell_metadata({"Square": {"x": 1, "y": 2, "z": 0}})
    assert meta["biome"] == "Forest"
    assert meta["terrain"] == "Grass"
